tests/test_paperscraper.py

```diff
@@ -7,27 +7,116 @@
 import paperscraper
 from paperscraper.exceptions import DOINotFoundError
 from paperscraper.headers import get_header
-from paperscraper.lib import clean_upbibtex, openaccess_scraper
-from paperscraper.utils import ThrottledClientSession
+from paperscraper.lib import (
+    clean_upbibtex,
+    doi_to_bibtex,
+    format_bibtex,
+    openaccess_scraper,
+    reconcile_doi,
+)
+from paperscraper.utils import ThrottledClientSession, find_doi
+
+
+class TestCrossref(IsolatedAsyncioTestCase):
+    async def test_reconcile_dois(self):
+        session = ThrottledClientSession(headers=get_header(), rate_limit=15 / 60)
+        link = "https://doi.org/10.1056/nejmoa2200674"
+        doi = "10.1056/nejmoa2200674"
+        assert find_doi(link) == doi
+
+        bibtex = await doi_to_bibtex(doi, session)
+        assert bibtex
+
+        # get title
+        title = bibtex.split("title={")[1].split("},")[0]
+        assert await reconcile_doi(title, [], session) == doi
+
+        # format
+        key = bibtex.split("{")[1].split(",")[0]
+        assert format_bibtex(bibtex, key, clean=False)
+
+
+def test_find_doi():
+    link = "https://www.sciencedirect.com/science/article/pii/S001046551930373X"
+    assert find_doi(link) is None
+
+    link = "https://doi.org/10.1056/nejmoa2200674"
+    assert find_doi(link) == "10.1056/nejmoa2200674"
+
+    link = "https://www.biorxiv.org/content/10.1101/2024.01.31.578268v1"
+    assert find_doi(link) == "10.1101/2024.01.31.578268v1"
+
+    link = "https://www.biorxiv.org/content/10.1101/2024.01.31.578268v1.full-text"
+    assert find_doi(link) == "10.1101/2024.01.31.578268v1"
+
+    link = "https://www.taylorfrancis.com/chapters/edit/10.1201/9781003240037-2/impact-covid-vaccination-globe-using-data-analytics-pawan-whig-arun-velu-rahul-reddy-pavika-sharma"
+    assert find_doi(link) == "10.1201/9781003240037-2"
```

Comment:
```suggestion
@pytest.mark.parameterize(("link", "expected"),
    [
        ("https://www.sciencedirect.com/science/article/pii/S001046551930373X", None),
        ("https://doi.org/10.1056/nejmoa2200674", "10.1056/nejmoa2200674"),
        ("https://www.biorxiv.org/content/10.1101/2024.01.31.578268v1", "10.1101/2024.01.31.578268v1"),
        ("https://www.biorxiv.org/content/10.1101/2024.01.31.578268v1.full-text", "10.1101/2024.01.31.578268v1"),
        ("https://www.taylorfrancis.com/chapters/edit/10.1201/9781003240037-2/impact-covid-vaccination-globe-using-data-analytics-pawan-whig-arun-velu-rahul-reddy-pavika-sharma", "10.1201/9781003240037-2"),
    ]
)
def test_find_doi(link: str, expected: str | None) -> None:
    if expected is None:
        assert find_doi(link) is None
    else:
        assert find_doi(link) == expected
```

`pytest.mark.parameterize` will make tests able to run in parallel, and also make it evident when multiple fail at once